File: c++/NeetCode150/9_longest_consecutive.cpp
#include <unordered_set>
#include <unordered_map>
#include <string>
#include <vector>
using namespace std;

class Solution {
public:
    int longestConsecutive(vector<int>& nums) 
    {
        unordered_set<int> numSet;
        int largest = 0;
        for(auto num: nums)
        {
           numSet.insert(num);
        }

        for(auto num: numSet)
        {
            if(numSet.count(num-1))
            {
                continue;
            }
            else
            {
                int size = 1;
                int new_num = num + 1;
                while(numSet.count(new_num))
                {
                    new_num++;
                    size++;
                }
                if(size > largest)
                {
                    largest = size;
                }
            }
        }
        return largest;
    }
};

File: java/NeetCode150/9_longest_consecutive.java
class Solution {
    public int longestConsecutive(int[] nums) 
    {
        // Create Set from array
        HashSet<Integer> numSet = new HashSet<Integer>();
        int largest = 0;
        for(int i=0;i<nums.length;i++)
        {
            numSet.add(nums[i]);
        }
        
        // Loop over array
        for(int i=0;i<nums.length;i++)
        {
            if(numSet.contains(nums[i] - 1))
            {
                continue;
            }
            else
            {
                int size = 1;
                int newNum = nums[i];
                while(numSet.contains(newNum + 1))
                {
                    size++;
                    newNum++;
                }
                if(size > largest)
                {
                    largest = size;
                }
            } 
        }
        return largest;
    }
}
